src/sys/socket/sockopt.rs

```diff
@@ -1727,15 +1775,14 @@ impl<'a> Set<'a, OsString> for SetOsString<'a> {
 }
 
 /// Getter for a `CString` value.
-#[cfg(apple_targets)]
-#[cfg(feature = "net")]
-struct GetCString<T: AsMut<[u8]>> {
+// Hide the docs, because it's an implementation detail of `sockopt_impl!`
+#[doc(hidden)]
+#[derive(Debug)]
+pub struct GetCString<T: AsMut<[u8]>> {
```

Comment:
Currently this `Get` impl is directly used in the `sockopt_impl!` implementation:

```rs
#[cfg(apple_targets)]
#[cfg(feature = "net")]
sockopt_impl!(
    /// Get the utun interface name.
    UtunIfname,
    GetOnly,
    libc::SYSPROTO_CONTROL,
    libc::UTUN_OPT_IFNAME,
    CString,
    GetCString<[u8; libc::IFNAMSIZ]> // NOTE
);
```

because we don't have such conversion branches for it:

```rs
macro_rules! sockopt_impl {
    ($(#[$attr:meta])* $name:ident, GetOnly, $level:expr, $flag:path, CString) => {
        sockopt_impl!($(#[$attr])*
                      $name, GetOnly, $level, $flag, bool, $crate::sys::socket::sockopt::GetCString);
    ($(#[$attr:meta])* $name:ident, SetOnly, $level:expr, $flag:path, CString) => {
        sockopt_impl!($(#[$attr])*
                      $name, GetOnly, $level, $flag, bool, $crate::sys::socket::sockopt::SetCString);
    };
```

So we should probably not hide it in the doc, but it is barely used, and we can fix this if becomes commonly used in the future so hiding it still looks good to me

Reply:
Oh, I guess I didn't need to make that one pub.